ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -917,6 +1038,48 @@ func (r *RayClusterReconciler) buildWorkerPod(instance rayv1alpha1.RayCluster, w
 	return pod
 }
 
+func (r *RayClusterReconciler) buildRedisCleanupJob(instance rayv1alpha1.RayCluster) batchv1.Job {
+	pod := r.buildHeadPod(instance)
+	pod.Labels[common.RayNodeTypeLabelKey] = string(rayv1alpha1.RedisCleanupNode)
+	// Only keep the Ray container in the Redis cleanup Job.
+	pod.Spec.Containers = []corev1.Container{pod.Spec.Containers[common.RayContainerIndex]}
+	pod.Spec.Containers[common.RayContainerIndex].Command = []string{"/bin/bash", "-lc", "--"}
+	pod.Spec.Containers[common.RayContainerIndex].Args = []string{
+		"python -c " +
+			"\"from ray._private.gcs_utils import cleanup_redis_storage; " +
```

Comment:
It's unfortunate that we have to use a private API for this.  Do you know if we test this codepath in the Kuberay repo against the Ray nightly wheels? That will help us catch the error if the private function changes its behavior or name

Reply:
Open a good first issue.
https://github.com/ray-project/kuberay/issues/1422